osbenchmark/synthetic_data_generator/strategies/mapping_strategy.py

```diff
@@ -6,33 +6,83 @@
+
+        self.logger = logging.getLogger(__name__)
+
+    def generate_data_chunks_across_workers(self, dask_client: Client, docs_per_chunk: int, seeds: list ) -> list:
+        """
+        Submits workers to generate data chunks and returns Dask futures
+
+        Returns: list of Dask Futures
+        """
+        futures = [dask_client.submit(self.generate_data_chunk_from_worker, docs_per_chunk, seed) for seed in seeds]
+
+        return futures
+
+    # pylint: disable=arguments-differ
+    def generate_data_chunk_from_worker(self, docs_per_chunk: int, seed: Optional[int]) -> list:
+        """
+        This method is submitted to Dask worker and can be thought of as the worker performing a job, which is calling the
+        MappingConverter's static method generate_fake_document() function to generate documents.
+        The worker will call the function N number of times to generate N docs of data before returning results.
+
+        Note: This method reconstructs the MappingConverter because Dask coordinator requires serializing and deserializing objects
+        when passing them to a worker. Generates the generate_fake_document, which gets invoked N number of times
+        before returning a list of documents.
+
+        Returns: List of generated documents.
+        """
+        # Initialize mapping generation values (params from sdg-config.yml) given to worker
+        mapping_generator_logic = MappingConverter(self.mapping_generation_values, seed)
+        mappings_with_generators = mapping_generator_logic.transform_mapping_to_generators(self.index_mapping)
+
+        documents = [MappingConverter.generate_fake_document(mappings_with_generators) for _ in range(docs_per_chunk)]
+
+        return documents
+
+    def generate_test_document(self):
+        mapping_converter = MappingConverter(self.mapping_generation_values)
+        converted_mappings = mapping_converter.transform_mapping_to_generators(self.index_mapping)
+
+        return MappingConverter.generate_fake_document(transformed_mapping=converted_mappings)
 
-class MappingSyntheticDataGenerator:
-    def __init__(self, mapping_config=None, seed=1):
+    def calculate_avg_doc_size(self):
+        # Didn't do pickle because this seems to be more accurate
+        output = [self.generate_test_document()]
+        write_chunk(output, '/tmp/test-size.json')
+
+        size = os.path.getsize('/tmp/test-size.json')
+        os.remove('/tmp/test-size.json')
+
+        return size
+
+class MappingConverter:
+    def __init__(self, mapping_generation_values=None, seed=1):
         self.logger = logging.getLogger(__name__)
-        # TODO: Set self.mapping_config to automatically point to MappingSyntheticDataGenerator
-        self.mapping_config = mapping_config or {}
+        self.mapping_config = mapping_generation_values if mapping_generation_values else {}
 
         self.generic = Generic(locale=Locale.EN)
```

Comment:
Is this tied to the English locale?  Perhaps document text could be in other languages as well.